### ISSUE TITLE
Reproduce the DCC01 effect

### ISSUE THREAD

**Issue by HR-zju** [HuanYin94/RaLL#3]
Hi, 
      I tried running DCC01 on your latest code, pre-trained model, map, pose file, and radar scan image produced by **RobotCar SDK**, but it didn't work well. Radar quickly lost its way. What's going on?
Greeting,
Sam

**Comment by HuanYin94** [comment]
> 

Hi, do you use odometry information?

**Comment by HR-zju** [comment]
> 

Thanks for your reply, 
Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?

**Comment by HuanYin94** [comment]
> > 
> 
> Thanks for your reply, Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?

single GPU is okay. I am not sure the modified loading method will have an effect or not.

**Comment by HR-zju** [comment]
> > > 
> > 
> > 
> > Thanks for your reply, Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?
> 
> single GPU is okay. I am not sure the modified loading method will have an effect or not.

I have changed following parts in test_ekf_loc_DCC01_NN.py
1.  data path(line.38-43,62,63)
2.  ` device = 'cuda:2'` ->     `device = 'cuda' `(line.82)
3. add `, map_location = device` at the end of `siamese_model.load_state_dict(torch.load(pre_sia_model_path)) `(line.95,106)
I think all these won't affect the results.

**Comment by HuanYin94** [comment]
> 



> > > > 
> > > 
> > > 
> > > Thanks for your reply, Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?
> > 
> > 
> > single GPU is okay. I am not sure the modified loading method will have an effect or not.
> 
> I have changed following parts in test_ekf_loc_DCC01_NN.py
> 
> 1. data path(line.38-43,62,63)
> 2. ` device = 'cuda:2'` ->     `device = 'cuda' `(line.82)
> 3. add `, map_location = device` at the end of `siamese_model.load_state_dict(torch.load(pre_sia_model_path)) `(line.95,106)
>    I think all these won't affect the results.

seems that these are not the reason. 
have you tried on other sequences in oxford dataset?

**Comment by HR-zju** [comment]
> > 
> 
> > > > > 
> > > > 
> > > > 
> > > > Thanks for your reply, Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?
> > > 
> > > 
> > > single GPU is okay. I am not sure the modified loading method will have an effect or not.
> > 
> > 
> > I have changed following parts in test_ekf_loc_DCC01_NN.py
> > 
> > 1. data path(line.38-43,62,63)
> > 2. ` device = 'cuda:2'` ->     `device = 'cuda' `(line.82)
> > 3. add `, map_location = device` at the end of `siamese_model.load_state_dict(torch.load(pre_sia_model_path)) `(line.95,106)
> >    I think all these won't affect the results.
> 
> seems that these are not the reason. have you tried on other sequences in oxford dataset?

Yes, I tried other sequences. But the result is basically a trajectory that keeps looping clockwise.

**Comment by HuanYin94** [comment]
> 



> > > 
> > 
> > 
> > > > > > 
> > > > > 
> > > > > 
> > > > > Thanks for your reply, Yes, I used DCC01_odom.txt. But I modified the loading method and ran it on a single GPU. Is there any problem with this?
> > > > 
> > > > 
> > > > single GPU is okay. I am not sure the modified loading method will have an effect or not.
> > > 
> > > 
> > > I have changed following parts in test_ekf_loc_DCC01_NN.py
> > > 
> > > 1. data path(line.38-43,62,63)
> > > 2. ` device = 'cuda:2'` ->     `device = 'cuda' `(line.82)
> > > 3. add `, map_location = device` at the end of `siamese_model.load_state_dict(torch.load(pre_sia_model_path)) `(line.95,106)
> > >    I think all these won't affect the results.
> > 
> > 
> > seems that these are not the reason. have you tried on other sequences in oxford dataset?
> 
> Yes, I tried other sequences. But the result is basically a trajectory that keeps looping clockwise.

What is the meaning of looping clockwise? Does the localization fail at the start point or  after traveling a distance? Could you plz describe with more details

**Comment by HR-zju** [comment]
Yes, it failed in the first place. But briefly, in oxford01, it works fine. After the positioning fails, the car will deflect to the right and circle clockwise.

Below is an example of the radar scan I generated(at the beginning of the DCC01). At the same time, I also tried to normalize the images, but I got the same result.

![2](https://user-images.githubusercontent.com/73371574/155712495-3d7d1aae-2646-4424-bdd4-4e0108418e43.png)

**Comment by HuanYin94** [comment]
> Yes, it failed in the first place. But briefly, in oxford01, it works fine. After the positioning fails, the car will deflect to the right and circle clockwise.
> 
> Below is an example of the radar scan I generated(at the beginning of the DCC01). At the same time, I also tried to normalize the images, but I got the same result.
> 
> ![2](https://user-images.githubusercontent.com/73371574/155712495-3d7d1aae-2646-4424-bdd4-4e0108418e43.png)

Do you mean that it only runs successfully on the whole session of seq01? but fails at the start position on the other sessions?
BTW, RaLL is proposed for pose tracking, not for global localization. If the pose error is larger than the bound (6m), it is hard to relocalize the robot.

**Comment by HuanYin94** [comment]
@HR-zju Please use an old version of pytorch.

I test it with 1.5.0 and it can run successfully.
